app/commands/decidim/participatory_documents/admin/update_or_create_annotation.rb

```diff
@@ -0,0 +1,96 @@
+# frozen_string_literal: true
+
+module Decidim
+  module ParticipatoryDocuments
+    module Admin
+      class UpdateOrCreateAnnotation < Rectify::Command
+        # Public: Initializes the command.
+        #
+        # form - A form object with the params.
+        def initialize(form, document)
+          @form = form
+          @document = document
+        end
+
+        def call
+          return broadcast(:invalid) if form.invalid?
+          return broadcast(:invalid) if document.nil?
+
+          begin
+            @old_section = annotation.section
+            transaction do
+              annotation.assign_attributes(**attributes)
+              create_annotation! if annotation.new_record?
+              create_section!
```

Comment:
```suggestion
              create_section! if section.blank?
              annotation.assign_attributes(**attributes)
              create_annotation! if annotation.blank?
```
I think we should create the section first (if does not exist). Take into account that an annotation can come grouped with another annotation that already has a section.
We should add a tests for these cases.

Then the accessors should be like:

```ruby
 def section
   @section ||= document.sections.find_by(id: form.section)
 end

 def annotation
   @annotation ||= document.annotations.find_by(id: form.id)
 end
```